### ISSUE TITLE
Error with mapnik-render and datasources plugins v3.x

### ISSUE THREAD

**Issue by samtux** [mapnik/mapnik#4010]
I always get the same error for the mapnik-render command. I have all the plugins installed but I can't find where the error occurs:

`% mapnik-render world.xml world.png`

`Error Could not create datasource for type: 'shape' (no datasource plugin directories have been successfully registered)  encountered during parsing of layer 'world' in Layer of 'world.xml'`

**Comment by talaj** [comment]
See https://github.com/mapnik/mapnik/issues/3373.

But feel free to extend it. It is very simple:
https://github.com/mapnik/mapnik/blob/7dcf5ee0826e41f3c4ae378dd094ad7942fb435d/utils/mapnik-render/mapnik-render.cpp#L106

I have [my own utility](https://github.com/talaj/mapnik-render) for rendering based on visual test runner from Mapnik. It has a lot of options: [src/run.cpp#L103-L141](https://github.com/talaj/mapnik-render/blob/ee3129d0056d3994ca8c41b5e2be87275ae2a55a/src/run.cpp#L103-L141).

**Comment by samtux** [comment]
Thanks for the answer @talaj . Is there any way to properly handle the register datasource path? Since the utility mapnik-render handles relative paths, and I must copy the plugins folder in the directory I'm performing the render.

The error persists in the development version (3.1.x) of Mapnik.

**Comment by talaj** [comment]
> Is there any way to properly handle the register datasource path?

A part of Mapnik installation is `mapnik-config` utility, which can provide various information about Mapnik. There is also information where plugins are installed:

```console
$ mapnik-config --input-plugins                                                                                                                                                                         
/usr/lib/mapnik/input
```

From my point of view, `mapnik-render` should contain a new command line argument for path of the directory with plugins. A user would be able to use it like this:

```sh
mapnik-render -plugins $(mapnik-config --input-plugins)
```

**Comment by larsga** [comment]
Yes, if one could specify the directory to `mapnik-render` that would solve the problem.

If `mapnik-render` could look for `mapnik-config` on the path that would in many cases solve the problem without the user having to do anything at all.
